src/components/OfferListTable.js

```diff
@@ -0,0 +1,97 @@
+import { useEffect, useState } from 'react';
+import { getitems, getitembyid, edititem, deleteitem } from '../APIs/ApiOffer';
+
+export default function OfferListTable({ isFilter, searchfilter, selectcategory }) {
+
+    const [offerList, setOfferList] = useState([]);
+
+    const loadData = () => {
+        getitems().then((response) => {
+            const data = response.data;
+            setOfferList(data);
+        }).catch(console.error());
+    }
+
+    useEffect(() => {
+        loadData();
+    }, []);
+
+    const toggleitems = (itemid) => {
+        getitembyid(itemid).then((response) => {
+            const toggle = response.data.isEnabled;
+            edititem(itemid, { isEnabled: !toggle }).then(() => { loadData() });
```

Comment:
show loader in all edit/get/delete/create operations

Reply:
Done